domain/infosync/info.go

```diff
@@ -271,6 +279,44 @@ func (is *InfoSyncer) RemoveServerInfo() {
 	}
 }
 
+type topologyInfo struct {
+	ServerVersionInfo
+	StatusPort uint
+	BinaryPath string
+}
+
+// RemoveServerInfo stores the topology of tidb to etcd.
+// Note: this part no ttl exists.
+func (is *InfoSyncer) storeTopologyInfo(ctx context.Context) error {
+	if is.etcdCli == nil {
+		return nil
+	}
+	// maybe it's not proper here.
+	s, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	topologyInfo := topologyInfo{
+		ServerVersionInfo: is.info.ServerVersionInfo,
+		StatusPort:        is.info.StatusPort,
+		BinaryPath:        s,
+	}
+
+	infoBuf, err := json.Marshal(topologyInfo)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	str := string(hack.String(infoBuf))
+	key := fmt.Sprintf("%s/%s:%v/tidb", TopologyInformationPath, is.info.IP, is.info.Port)
+	// Note: no lease is required here.
+	err = util.PutKVToEtcd(ctx, is.etcdCli, keyOpDefaultRetryCnt, key, str)
```

Comment:
with lease will be better?
If tidb panic unexpected, the info won't be deleted.

Reply:
If it's not ttl session, we can just leave it here.

Reply:
I mean, if TiDB panic, the KV will never be deleted, except user manual delete.

Reply:
(Sure, It was designed like this, user can delete non-ttl key in tidb-dashboard)